### PR TITLE
TVS2: README, docs and site

### LINKED ISSUE #2018: Update Architecture / Nessie Spec / Commit Kernel documentation after TVS2 is done

### PR DESCRIPTION
**Pull request by snazy**
Depends on #2010 + #2083 + #2084 + #2085

Fixes #2018 

---
This change is [Reviewable](https://reviewable.io/reviews/projectnessie/nessie/2086)

### CODE REVIEW

File: site/docs/develop/kernel.md [hunk 1]
```diff
@@ -1,47 +1,264 @@
...
+Implementations are based
+on [NonTransactionalDatabaseAdapter](https://github.com/projectnessie/nessie/blob/main/versioned/persist/nontx/src/main/java/org/projectnessie/versioned/persist/nontx/NonTransactionalDatabaseAdapter.java)
+and only implement the database specific "primitives" to unconditionally read and write records and
+perform the mandatory CAS (compare-and-swap) operation.
+
+Key-value stores are all non-transactional as those are built for scale-out. Most key-value stores
+support atomic CAS (compare-and-swap) operations against a single row/record, but atomic and
+conditional updates to multiple rows/records is either not supported at all or extremely slow.
+
+Within Nessie we differ between content-types that do require so called "global-state" and those
```
Comment (nastra) [inline site/docs/develop/kernel.md:183]: we differ -> we differentiate

File: site/docs/develop/kernel.md [hunk 2]
```diff
@@ -1,47 +1,264 @@
...
+* Concurrent commits against the same table (think: Iceberg or Deltalake table) are slower than
+  concurrent commits against different tables.
+
+### Gatling Benchmarks
+
+Nessie has a framework to simulate "higher level use cases" using Gatling. See the readmes
+[here](https://github.com/projectnessie/nessie/blob/main/perftest/gatling/README.md) and
+[here](https://github.com/projectnessie/nessie/blob/main/perftest/simulations/README.md). Please
+note that all kinds of performance tests are only meaningful in production-like environments using
+production-live use cases.
```
Comment (nastra) [inline site/docs/develop/kernel.md:249]: production-live -> production-like

File: site/docs/develop/kernel.md [hunk 3]
```diff
@@ -1,47 +1,264 @@
...
+
+### Microbenchmarks
+
+There are microbenchmarks available, which can be useful to investigate the overall performance of a
+database. Please note that performance tests, even microbenchmarks, are only meaningful in
+production-like environments using production-live use cases.
```
Comment (nastra) [inline site/docs/develop/kernel.md:255]: production-live -> production-like

**Review by nastra — CHANGES_REQUESTED** [review]

File: site/docs/develop/nessie_vs_git.md [hunk 4]
```diff
@@ -10,8 +11,8 @@ copies. This allows several other dimensions to be substantially more powerful.
...
-|Scale|100s MB|Unconstrained|While there are multiple examples of larger or higher performance Git implementations [[1](https://docs.microsoft.com/en-us/azure/devops/learn/git/technical-scale-challenges), [2](https://medium.com/palantir/stemma-distributed-git-server-70afbca0fc29)] , in general Git repositories are fairly small in size. Things like Git LFS were created to help accommodate this but given the nature of clones, large repositories are frowned upon. Because Nessie provides a centralized repository, typical repository constraints do not apply.
+|Speed (commits/second))|<1|hundreds/thousands|When we started working on Nessie, we actually tried to use Git. We evaluated Git directly, implemented a version that used JGit (used by tools like Gerrit and Eclipse) as well as explored the capabilities of GitHub, Azure Git and AWS Git. What we saw was a fairly expensive operation. Typically, a single commit operation took on the order of a few seconds.|
```
[the remark this answers is not in the document]
Comment (nastra) [reply site/docs/develop/nessie_vs_git.md:14]: or should this say `hundreds to thousands commits/second`?

File: site/docs/develop/kernel.md [hunk 5]
```diff
@@ -1,47 +1,264 @@
...
-At a high level, Nessie's commit kernel breaks keyspace into 
-a 3 level tree and each operation restates some portion of that tree. Each level of 
-the tree is identified by sha256 20 byte hash value, which is used to as a storage 
-key in DynamoDB. Commit operations are atomic and interact with branch objects using 
-a data structure akin to a 151-way striped locked. This data structure is updated using 
-patch conditional operations against DynamoDB. To maintain linear history, the 
-branch object also maintains a commit log that works with the striped lock to make 
-history clear and consistent. 
+`IcebergTable` contains the _current_ and _global_ pointer to Iceberg's table-metadata plus the
+ID of the snapshot defined in the table-metadata. Since Iceberg's table-metadata manages information
```
[the remark this answers is not in the document]
Comment (snazy) [reply site/docs/develop/kernel.md:31]: It's rather a term than a concrete interface or class here.

File: site/docs/develop/kernel.md [hunk 6]
```diff
@@ -1,47 +1,264 @@
...
+that must be consistent across all branches in Nessie, it is stored as so-called "global state".
+The value of the snapshot-ID is stored per Nessie named-reference (branch or tag).
 
+Updating _global-state_ and _on-reference-state_ are technically operations against two different
```
[the remark this answers is not in the document]
Comment (snazy) [reply site/docs/develop/kernel.md:35]: This one references the "table name"

File: site/docs/develop/kernel.md [hunk 7]
```diff
@@ -1,47 +1,264 @@
...
+key-value stores and transactional databases. Support for non-transactional databases, the
+data-model, is designed in a way that only requires a single CAS-operation to ensure atomicity and
+consistency even when committing two logical entities, namely the _global-state_ and the
+_on-reference-state_, respectively the update to the "HEAD" of the updated branch. Some more details
+are outlined below.
+
+### Version-Store and Database-Adapters
+
+* REST-API
+  * `VersionStore` interface defining the contract for the REST-API, deals with concrete contents
```
[the remark this answers is not in the document]
Comment (snazy) [reply site/docs/develop/kernel.md:50]: Updated

File: site/docs/develop/kernel.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (nastra) [reply site/docs/develop/kernel.md:31]: right, then I would just rename to table metadata without the "-"

[the remark this answers is not in the document]
Comment (nastra) [reply site/docs/develop/kernel.md]: there are still 3 occurrences of table-metadata in the file btw

File: site/docs/develop/kernel.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (nastra) [reply site/docs/develop/kernel.md:35]: not sure what you mean by that comment? can you elaborate? It's not clear to the reader imo why formatting is inconsistent across those terms

File: site/docs/develop/nessie_vs_git.md [hunk 8]
```diff
@@ -10,8 +11,8 @@ copies. This allows several other dimensions to be substantially more powerful.
 |Dimension|Git|Nessie|Rationale|
 |---|---|---|---|
 |Clones|Allowed|Not Allowed|This is the biggest difference between Nessie and Git. Git is a distributed version control system, Nessie is not. This is appropriate in the context of Nessie's role as an [RPS](../tables/index.md#root-pointer-store). When talking about Cloud Data ops, everyone doesn’t get their own copy of data--the datasets are typically large and centralized. Because Nessie is layered on top of those shared datasets, it clones make less sense. In the Nessie world, using personal branches provides a similar mechanism while keeping a shared world view of what can be managed for GC policies, etc.|
```
[the remark this answers is not in the document]
Comment (dimas-b) [reply site/docs/develop/nessie_vs_git.md]: +1 TBH, I do not understand what that phase was meant to say

File: site/docs/develop/kernel.md [hunk 9]
```diff
@@ -1,47 +1,268 @@
...
-are not done on this table except in the case of rare Garbage Collection operations. 
+Contention by itself is not avoidable, because operations against Nessie are guaranteed to be
+atomic and consistent.
 
+### Nessie Content-Types
 
-Space Consumption: The data stored in DynamoDB is designed to largely fit within DynamoDBs 4kb read size 
-unit (and 4x the write size unit). This true for objects in both the refs table and 
-the objects table.
+The state of so called `Contents` objects like `IcebergTable` or `DeltaLakeTable` represent the
```
Comment (dimas-b) [inline site/docs/develop/kernel.md:25]: `represents` (with an `s`)

File: site/docs/develop/kernel.md [hunk 10]
```diff
@@ -1,47 +1,268 @@
...
+pair. This means, that the data-model has to be fundamentally different for non-transactional
+key-value stores and transactional databases. Support for non-transactional databases, the
+data-model, is designed in a way that only requires a single CAS-operation to ensure atomicity and
+consistency even when committing two logical entities, namely the _global-state_ and the
+_on-reference-state_, respectively the update to the "HEAD" of the updated branch. Some more details
+are outlined below.
+
+### Version-Store and Database-Adapters
+
+Nessie's REST-API implementation works against the `VersionStore` interface, which defines the
```
Comment (dimas-b) [inline site/docs/develop/kernel.md:49]: tangential: Do we need `VersionStore` now that we have `DatabaseAdapter`? Do we have `VersionStore` implementations that are not based on a `DatabaseAdapter`?

File: site/docs/try/releases.md [hunk 11]
```diff
@@ -1,5 +1,9 @@
 # Releases
 
+## 0.10.0 Release (August xx, 2021)
+
+* New tiered-version-store implementation
```
Comment (dimas-b) [inline site/docs/try/releases.md:5]: nit: is it still tiered?

File: python/HISTORY.rst [hunk 12]
```diff
@@ -2,6 +2,11 @@
 History
 =======
 
+0.10.0 (2021-09-xx)
```
[the remark this answers is not in the document]
Comment (snazy) [reply python/HISTORY.rst:5]: Rather 1.0 though

File: site/docs/develop/kernel.md [hunk 13]
```diff
@@ -1,47 +1,268 @@
 # Commit Kernel
 
-Nessie's production commit kernel is optimized to provide high commit 
-throughput against a distributed key value store that provides record-level ACID 
-guarantees. Today, this kernel is built on top of DynamoDB. The commit kernel is the 
-heart of Nessie's oeprations and enables it to provide lightweight creation of new 
-tags and branches, merges, rebases all with a very high concurrent commit rate. 
+Nessie's production commit kernel is optimized to provide high commit throughput against a
+distributed key value store that provides record-level compare-and-swap capability or
+transactional/relational databases. The commit kernel is the heart of Nessie's operations and
```
[the remark this answers is not in the document]
Comment (snazy) [reply site/docs/develop/kernel.md:5]: That's why there are two different implementations.

File: site/docs/develop/kernel.md [hunk 14]
```diff
@@ -1,47 +1,268 @@
...
+enables it to provide lightweight creation of new tags/branches, merges, and rebases, all with very
+high concurrent commit rate.
 
-## Data Structures
+## High level abstract
 
-Nessie's commit kernel works with two tables: the refs table and the objects table. 
+Nessie 1.0 comes with a version-store (aka commit-kernel) implementation that is different from both
```
[the remark this answers is not in the document]
Comment (snazy) [reply site/docs/develop/kernel.md:11]: To my limited english knowledge, it all depends on what people agree on :disappear:

File: site/docs/develop/kernel.md (hunk 10 above)
Comment (snazy) [reply site/docs/develop/kernel.md:49]: VS knows about the concrete types. DataabseAdapter does not.

File: site/docs/try/releases.md (hunk 11 above)
Comment (snazy) [reply site/docs/try/releases.md:5]: We'll re-work the release-notes for the next release anyway.

File: site/docs/develop/kernel.md (hunk 13 above)
[the remark this answers is not in the document]
Comment (dimas-b) [reply site/docs/develop/kernel.md:5]: Maybe, `Nessie's production commit kernel has different modules to provide high ...`.

In my reading `kernel is optimized` kind of implies a monolithic implementation that handles all cases... but this is really nit-picky... I do not insist.

File: site/docs/develop/kernel.md (hunk 14 above)
[the remark this answers is not in the document]
Comment (nastra) [reply site/docs/develop/kernel.md:11]: having hyphens there feels wrong and looking up when to use hyphens is basically what @dimas-b said.

File: site/docs/develop/kernel.md [hunk 15]
```diff
@@ -1,108 +1,108 @@
 # Commit Kernel
 
 Nessie's production commit kernel is optimized to provide high commit throughput against a
-distributed key value store that provides record-level compare-and-swap capability or
+distributed key value store that provides record level CAS (compare and swap) capability or
```
Comment (dimas-b) [inline site/docs/develop/kernel.md]: sorry about nit-picking, but in this case, I think, hyphens were justified

**Review by dimas-b — APPROVED** [review]

**Review by dimas-b — APPROVED** [review]

**Review by nastra — APPROVED** [review]

**Pull request merged**